### PR TITLE
IndexedDB: Tests for IDBKeyRange and IDBRequest exceptions

### PR DESCRIPTION
**Pull request by inexorabletash**
A few cases not handled by existing tests, as identified by code
coverage analysis.

* IDBKeyRange.only()/lowerBound()/upperBound() with invalid keys
* IDBRequest.result/error getters accessed while request was pending

### CODE REVIEW

**Comment by inexorabletash** [comment]
@pwnall - can you review?

File: IndexedDB/idbkeyrange.htm [hunk 1]
```diff
@@ -32,6 +40,14 @@
         assert_false(keyRange.lowerOpen, "keyRange.lowerOpen");
     }, "IDBKeyRange.lowerBound() - 'open' parameter has correct default set");
 
+    test( function() {
+        assert_throws('DataError', function() { IDBKeyRange.lowerBound(undefined); }, 'undefined is not a valid key');
+        assert_throws('DataError', function() { IDBKeyRange.lowerBound(null); }, 'null is not a valid key');
+        assert_throws('DataError', function() { IDBKeyRange.lowerBound({}); }, 'Object is not a valid key');
+        assert_throws('DataError', function() { IDBKeyRange.lowerBound(Symbol()); }, 'Symbol is not a valid key');
+        assert_throws('DataError', function() { IDBKeyRange.lowerBound(true); }, 'boolean is not a valid key');
+    }, "IDBKeyRange.lowerBound() - throws on invalid keys");
```
Comment (pwnall) [inline IndexedDB/idbkeyrange.htm]: Same comment re: functions here and below.

File: IndexedDB/idbrequest_result.html [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+<script>
+async_test(function(t) {
+  var open = createdb(t);
+  open.onupgradeneeded = t.step_func(function(e) {
+    var db = e.target.result;
+    db.createObjectStore('store');
+  });
+  open.onsuccess = t.step_func(function(e) {
+    var db = e.target.result;
+    var request = db.transaction("store").objectStore("store").get(0);
```
Comment (pwnall) [inline IndexedDB/idbrequest_result.html]: double quotes are inconsistently used here, compared to the rest of the file

File: IndexedDB/idbrequest_error.html [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+<script>
+async_test(function(t) {
+  var open = createdb(t);
+  open.onupgradeneeded = t.step_func(function(e) {
+    var db = e.target.result;
+    db.createObjectStore('store');
+  });
+  open.onsuccess = t.step_func(function(e) {
+    var db = e.target.result;
+    var request = db.transaction("store").objectStore("store").get(0);
```
Comment (pwnall) [inline IndexedDB/idbrequest_error.html]: double quotes are inconsistently used here, compared to the rest of the file

**Comment by pwnall** [comment]
@inexorabletash LGTM, modulo nits. I think that the new files would be a tad more readable if you used arrow functions, but I'd be OK with having them merged as they are right now.

**Comment by inexorabletash** [comment]
Thanks @pwnall - I think I got all of those. Can you double check?

File: IndexedDB/idbrequest_error.html [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    var db = e.target.result;
+    db.createObjectStore('store');
+  });
+  open.onsuccess = t.step_func(e => {
+    var db = e.target.result;
+    var request = db.transaction('store').objectStore('store').get(0);
+
+    assert_equals(request.readyState, 'pending');
+    assert_throws('InvalidStateError',
+                  function() { request.error; },
```
Comment (pwnall) [inline IndexedDB/idbrequest_error.html]: Missed a `function` here? I think you might be able to collapse the line into the previous one if you use the arrow notation.

File: IndexedDB/idbrequest_result.html [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    var db = e.target.result;
+    db.createObjectStore('store');
+  });
+  open.onsuccess = t.step_func(e => {
+    var db = e.target.result;
+    var request = db.transaction('store').objectStore('store').get(0);
+
+    assert_equals(request.readyState, 'pending');
+    assert_throws('InvalidStateError',
+                  function() { request.result; },
```
Comment (pwnall) [inline IndexedDB/idbrequest_result.html]: also missed a `function` here

**Pull request merged**